app/controllers/requests_controller.rb

```diff
@@ -0,0 +1,80 @@
+class RequestsController < ApplicationController
+  # TODO: Remove me, should be set somewhere else
+  before_action :set_current_user
+
+  def index
+    if @current_user.is_supervisor
+      @requests = Request.all
+    else
+      @requests = Request.where(user: User.where(is_supervisor: false))
+    end
+  end
+
+  def new
+    @request = Request.new
+    @request_types = RequestType.all
+  end
+
+  def create
+    @request = Request.new(request_save_params)
+    date_format = '%m/%d/%Y %I:%M %p'
+    @request.start_time = Date.strptime(request_save_params[:start_time], date_format)
+    @request.end_time = Date.strptime(request_save_params[:end_time], date_format)
+    @request.user = @current_user
+
+    if @request.save
+      flash[:success] = 'Request Successfully Created'
+      redirect_to requests_path
+    else
+      flash[:warning] = @request.errors.full_messages
+      redirect_to action: :new
+    end
+  end
+
+  def edit
+    @request = Request.find(params[:id])
+    if can_modify_request
+      flash[:danger] = 'You do not have permission to edit a supervisor request'
+      redirect_to action: :index
+      return
+    end
+    @request_types = RequestType.all
+  end
+
+  def update
+    @request = Request.find(params[:id])
+    if can_modify_request
+      flash[:danger] = 'You do not have permission to update a supervisor request'
+      redirect_to action: :index
```

Comment:
I knew we had done it in the past, but then it wasn't working cause I was using `&&` instead of `and`, so I figured it made more sense to just put it on the next line to avoid my confusion.

Reply:
😆